Src/PerformanceCollector/Shared/Implementation/PerformanceCounterData.cs

```diff
@@ -17,11 +17,27 @@ internal class PerformanceCounterData
             this.UsesInstanceNamePlaceholder = usesInstanceNamePlaceholder;
             this.IsCustomCounter = isCustomCounter;
             this.IsInBadState = isInBadState;
```

Comment:
It's defined based on the first collection attempt - if first read failed then counter is in the bad state and nobody will try to read it again.